### PR TITLE
[typescript] migrate url-parse to URL WHATGW in https.ts

### PR DESCRIPTION
**Pull request by davidgamero**
Start moving away from the url-parse package in favor of the [WHATWG url api](https://nodejs.org/api/url.html#the-whatwg-url-api)
This follows convention and will make future implementations towards the fetch api smoother

from url-parse [npm page](https://www.npmjs.com/package/url-parse):
> url-parse was created in 2014 when the WHATWG URL API was not available in Node.js and the URL interface was supported only in some browsers. Today this is no longer true. The URL interface is available in all supported Node.js release lines and basically all browsers. Consider using it for better security and accuracy.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript/http/http.mustache [hunk 1]
```diff
@@ -2,7 +2,7 @@
 {{#node}}
 // TODO: evaluate if we can easily get rid of this library
 import {{^supportsES6}}* as{{/supportsES6}} FormData from "form-data";
-import { URLSearchParams } from 'url';
+import { URL, URLSearchParams } from 'url';
 import * as http from 'http';
 import * as https from 'https';
 {{/node}}
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript/http/http.mustache:8]: Can the code below be removed?
https://github.com/OpenAPITools/openapi-generator/blob/9fe05a4c87357bca3e1e45226d088bea0632e1b9/modules/openapi-generator/src/main/resources/typescript/http/http.mustache#L10-L14

and 

https://github.com/OpenAPITools/openapi-generator/blob/9fe05a4c87357bca3e1e45226d088bea0632e1b9/modules/openapi-generator/src/main/resources/typescript/http/http.mustache#L60-L96

**Review by macjohnny — COMMENTED** [review]
Thanks for your contribution!

File: modules/openapi-generator/src/main/resources/typescript/http/http.mustache (hunk 1 above)
Comment (davidgamero) [reply modules/openapi-generator/src/main/resources/typescript/http/http.mustache:8]: yep!

**Comment by davidgamero** [comment]
not totally sure what happened, but the python fastapi test is failing now
i can repro locally with 
`mvn integration-test -f samples/server/petstore/python-fastapi`

**Comment by macjohnny** [comment]
@adatta02 thanks for reporting, do you want to fix it?

**Comment by adatta02** [comment]
@macjohnny sure, happy to give it a shot.

**Comment by adatta02** [comment]
@macjohnny do you think it would be to intrusive to add an optional "baseUrl" parameter to the [ServerConfiguration](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript/http/servers.mustache) and [RequestContext](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript/http/http.mustache#L85) constructors?

**Comment by bodograumann** [comment]
This should fix your use case, @adatta02 : https://github.com/OpenAPITools/openapi-generator/pull/15482

**Pull request merged**